spec/unit/ssl/validator_spec.rb

```diff
@@ -236,149 +246,41 @@
   describe '#has_authz_peer_cert' do
     context 'when the Root CA is listed as authorized' do
       it 'returns true when the SSL cert is issued by the Master CA' do
-        expect(subject.has_authz_peer_cert(cert_chain, [root_ca_cert])).to be_truthy
+        expect(subject.has_authz_peer_cert(cert_chain, [pki[:root_cert]])).to be_truthy
       end
 
       it 'returns true when the SSL cert is issued by the Agent CA' do
-        expect(subject.has_authz_peer_cert(cert_chain_agent_ca, [root_ca_cert])).to be_truthy
+        expect(subject.has_authz_peer_cert(cert_chain_alternate, [pki[:root_cert]])).to be_truthy
       end
     end
 
-    context 'when the Master CA is listed as authorized' do
-      it 'returns false when the SSL cert is issued by the Master CA' do
-        expect(subject.has_authz_peer_cert(cert_chain, [master_ca_cert])).to be_truthy
+    context 'when one intermediate CA is listed as authorized' do
+      it 'returns true when the SSL cert is issued by the same intermediate CA' do
+        expect(subject.has_authz_peer_cert(cert_chain, [pki[:int_cert]])).to be_truthy
       end
 
-      it 'returns true when the SSL cert is issued by the Agent CA' do
-        expect(subject.has_authz_peer_cert(cert_chain_agent_ca, [master_ca_cert])).to be_falsey
+      it 'returns false when the SSL cert is issued by a different intermediate CA' do
+        expect(subject.has_authz_peer_cert(cert_chain_alternate, [pki[:int_cert]])).to be_falsey
       end
     end
 
-    context 'when the Agent CA is listed as authorized' do
-      it 'returns true when the SSL cert is issued by the Master CA' do
-        expect(subject.has_authz_peer_cert(cert_chain, [agent_ca_cert])).to be_falsey
+    context 'when another intermediate CA is listed as authorized' do
```

Comment:
Is there any value in this second set of tests? It seems like it's testing the same behavior as the first?

Reply:
I don't think so; I just did it to have parity with what was here before, but I also think it can be 🔪.